staging/src/k8s.io/client-go/dynamic/simple.go

```diff
@@ -32,8 +32,7 @@ import (
 )
 
 type DynamicInterface interface {
-	ClusterResource(resource schema.GroupVersionResource) DynamicResourceInterface
-	NamespacedResource(resource schema.GroupVersionResource, namespace string) DynamicResourceInterface
```

Comment:
These two Resource related interface has been renamed to Resource. Then I doubt why not also rename the below subresource interface.

Reply:
> These two Resource related interface has been renamed to Resource. Then I doubt why not also rename the below subresource interface.

It's not obvious what the future of subresources via this API is.  A lot depends on what we consider the primary subresources use-cases to be.  That's part of why the methods are deprecated.  I'm actually not sure we use them anywhere.

Reply:
One [request](https://github.com/kubernetes/kubernetes/issues/49588) for supporting subresources in dynamic client was to facilitate e2e testing. E.g. the [API coverage test](https://github.com/kubernetes/kubernetes/pull/57171) uses this.